features/0075-payment-decorators/README.md

```diff
@@ -0,0 +1,335 @@
+# 00xx: Payment Decorators
+- Author: Sam Curren (sam@sovrin.org), Daniel Hardman (daniel.hardman@gmail.com), Tomislav Markovski
+- Start Date: 2019-04-22
+
+## Status
+- Status: [PROPOSED](/README.md#rfc-lifecycle)
+- Status Date: 2019-06-11
+- Status Note: Still being studied. This supersedes [Indy HIPE PR #129](
+https://github.com/hyperledger/indy-hipe/pull/129).
+
+## Summary
+[summary]: #summary
+
+Defines the `~payment_request`, `payment_internal_response`, and `~payment_receipt` decorators.
+These offer standard payment features in all DIDComm interactions, and let DIDComm take advantage
+of the [W3C's Payment Request API]( https://www.w3.org/TR/payment-request) in an interoperable way.
+
+## Motivation
+[motivation]: #motivation
+
+Instead of inventing custom messages for payments in each protocol, arbitrary messages
+can express payment semantics with payment decorators. Individual protocol specs should
+clarify on which messages and under which conditions the decorators are used.
+
+## Tutorial
+
+The [W3C's Payment Request API](https://www.w3.org/TR/payment-request) governs interactions
+between three parties:
+
+1. payer
+2. payee
+3. payment method
+
+The payer is usually imagined to be a person operating a web browser,
+the payee is imagined to be an online store, and the payment method might be something like
+a credit card processing service. The payee emits a [PaymentRequest](
+https://www.w3.org/TR/payment-request/#paymentrequest-interface) JSON structure (step 1
+below); this causes the payee to be prompted (step 2, "Render"). The payer decides
+whether to pay, and if so, which payment method and options she prefers (step 3, "Configure").
+The payer's choices are embodied in a [PaymentResponse](
+https://www.w3.org/TR/payment-request/#paymentresponse-interface)
+JSON structure (step 4). This is then used to select the appropriate codepath and inputs to
+invoke the desired payment method (step 5).
+
+[![API flow](payment-request-api-flow.png)](payment-request-api-flow.puml)
```

Comment:
@dhh1128 the PaymentResponse is sent back to the payee after the user has made a payment

Reply:
@llorllale That was my original understanding, too. However, when I went and studied the spec carefully, I could not find any mention of that. And the one good example that they give of a PaymentResponse, BasicCardResponse, contains card info that the payee shouldn't have if they're not processing a payment. So I came to doubt my first intuition. Can you point me to a section of the spec that clarifies?

Reply:
I'm going to approve this so we can get this into the repo. Let's move this conversation to the issues section.